lsp-completion.el

```diff
@@ -525,7 +525,15 @@ The MARKERS and PREFIX value will be attached to each candidate."
        :company-require-match 'never
        :company-prefix-length
        (save-excursion
-         (and (lsp-completion--looking-back-trigger-characterp trigger-chars) t))
+         (defvar company-minimum-prefix-length)
+         (let ((bounds-left (max (line-beginning-position) (- (point) company-minimum-prefix-length)))
```

Comment:
I don't think you need `company-minimum-prefix-length` here at all.

Just see what `company-clang--prefix` does, for example: https://github.com/company-mode/company-mode/blob/master/company-clang.el#L336-L339

As a backend, you have two choices: either return a number (the real of "virtual" prefix length), or `t`, if the current prefix is preceded by a prefix character such as `.`. The framework itself will then do the prefix length check if needed (if you returned a number and not `t`).

Reply:
Yes, I do not need to use `company-minimum-prefix-length` and can just keep looking back until the begin of line or the first whitespace character, however using the `company-minimum-prefix-length` is better as it can be quite short and doesn't affect the perf much. I will use other mean to get the approximated length to check then.

Reply:
You could just limit the search by the beginning of the current line. Or the first whitespace char (like is already done). This is fast enough not to require additional performance tweaks, IMO.